DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+        _
+    }
+    
+    function Deed(address _bidder, bytes32 _sealedBid, address _registrar) {
+        bidder = _bidder;
+        registrar = Registrar(_registrar);
+        sealedBid = _sealedBid;
+        creationDate = now;
+    }
+    
+    function unsealBid(bytes32 _hashedName, address _owner, uint _value, bytes32 _salt) noEther onlyRegistrar {
+        if (sha3(_hashedName, _owner, _value, _salt) != sealedBid) throw;
+        hashedName = _hashedName;
+        owner = _owner;
+        value = _value;
+        if (!bidder.send(this.balance - _value)) throw;
+    }
+    
+    function closeDeed( uint burnAmount) noEther onlyRegistrar {
+        burn.send((burnAmount * this.balance)/1000);
+        selfdestruct(bidder);
+    }
+    
+    function setBalance(uint newValue) onlyRegistrar {
+        // Check if it has enough balance to set the value
+        if (this.balance < newValue) throw;
+        // Send the difference to the owner
+        if (!owner.send(this.balance - newValue)) throw;
+    }
+    
+    function transferOwnership(address newOwner) {
+        if (msg.sender != owner ) throw;
+        owner = newOwner;
+    }
+}
+
+contract Registrar {
+    mapping (bytes32 => entry) public hashedNames;
+    mapping (bytes32 => address) public sealedBids;
+    
+    enum Mode { Open, Auction, Owned }
+    uint auctionLength = 20 minutes;
+    uint revealPeriod = 5 minutes;
+    uint renewalPeriod = 2 hours;
+    uint M = 1000000; // just a multiplier to get more precision on averages
+    uint public averagePeriod;
+    uint public lastSinceNewName;
+    uint public minBid = 0.1 ether;    // TODO: figure out a better way to come up with this
+    uint public averagePrice = minBid;
+    
+    struct entry {
+        Mode status;
+        address deed;
+        uint auctionDate;
+        uint value;
+        uint highestBid;
+        uint firstRegistered;
+        uint lastRenewed;
+        uint renewalDate;
+        uint averagePrice;
+    }
+    
+    // doesnt work
+    modifier withName(bytes32 nameHash){
+        entry name = hashedNames[nameHash];
+        _        
+    }
+    
+    modifier noEther  {
+        if (msg.value > 0) throw;
+        _
+    }   
+    
+    function Registrar() noEther  {
+        lastSinceNewName = now;
+    }
+
+    /*
+    ## Start Auction for available name
+
+    Anyone can start an auction by sending an array of hashes that they want to bid for. 
+    Arrays are sent so that someone can open up an auction for X dummy hashes when they 
+    are only really interested in bidding for one. This will increase the cost for an 
+    attacker from simply bidding on all new auctions blindly. Dummy auctions that are 
+    open but not bid on are closed after a week. 
+    */    
+    function openAuctions(bytes32[] auctionsArray) noEther{
+
+        for (uint i = 0; i < auctionsArray.length; i ++ ) {
+            entry newAuction = hashedNames[auctionsArray[i]];
```

Comment:
Could we call this `entry` or `registration`? It's not yet clear that it _is_ a new auction, which makes some of the following code a bit hard to understand.